### DISCUSSION TITLE
Shall we generate converters for structural types

### DISCUSSION THREAD

**Discussion by Atry** [ScalablyTyped/Converter#382] [Ideas]
Suppose we have a type defined in TypeScript like this:
```scala
@js.native
trait MyType extends StObject {
  val field: Double = js.native
}
```
The idea is to also generate the following helper:

```scala
object MyType {
  @inline implicit def isStructural = <:<.refl[MyType].asInstanceOf[js.Any { val field: Double } =:= MyType]
}
```

With the help of the `=:=` instance, other JS types that contain a `field: Double` can be converted to `MyType` safely.

**Comment by oyvindberg** [comment]
Haha wow! that's rather interesting!

I honestly have no idea. I've been avoiding structural types throughout my adventures through Scala. 

How useful will this be? is there a subtype relationship between structural types where one is a subset of the other? would this be transitive between types?

Do you know how and if `Selectable` in Scala 3 would apply here?

**Reply by Atry** [reply]
> How useful will this be? is there a subtype relationship between structural types where one is a subset of the other? 

`<:<` and `=:=` are type classes in the Scala library to prove sub-typing, but they are not actually sub-typing from the compiler's point of view. So it only helps the user to convert between types safely, instead of affecting the compiler behavior.

> would this be transitive between types?

It would require the user to call `<:<.compose` manually to create transitive instances.
